src/test/unit/algorithms/ConnectionsPerformanceTest.cpp

```diff
@@ -78,22 +83,26 @@ float runTemporalMemoryTest(UInt numColumns, UInt w,   int numSequences,
     for (auto sequence : sequences) {
       for (auto sdr : sequence) {
         tm.compute(sdr, true);
-	//TODO get untrained anomaly score here
+	avgAnomAfter = anom10.compute(tm.anomaly); //average anomaly score
       }
       tm.reset();
     }
+    NTA_CHECK(avgAnomBefore >= avgAnomAfter) << "TM should learn and avg anomalies improve, but we got: "
+      << avgAnomBefore << " and now: " << avgAnomAfter; //invariant
+    avgAnomBefore = avgAnomAfter; //update
```

Comment:
implement the anomaly checks whether TM learns, as you've wanted long ago @ctrl-z-9000-times